src/status_im/ui/screens/wallet/settings/events.cljs

```diff
@@ -0,0 +1,18 @@
+(ns status-im.ui.screens.wallet.settings.events
+  (:require [status-im.utils.ethereum.core :as ethereum]
+            [status-im.utils.handlers :as handlers]))
+
+(defn- mark-checked [ids id checked?]
+  (if checked?
+    (conj ids id)
+    (disj ids id)))
+
+(handlers/register-handler-fx
+  :wallet.settings/toggle-visible-token
+  (fn [{{:keys [network] :accounts/keys [current-account-id] :as db} :db} [_ symbol checked?]]
+    (let [chain-id     (ethereum/network->chain-id network)
+          path         [:accounts/accounts current-account-id :settings]
+          settings     (get-in db path)
+          new-settings (update-in settings [:wallet :visible-tokens chain-id] #(mark-checked % symbol checked?))]
+      {:db       (assoc-in db path new-settings)
+       :dispatch [:account-update-wallet-settings new-settings]})))
```

Comment:
Why is this an event ? No reason for that, it should be just function taking anything necessary (fx, cofx, or both) and creating a new effects map. 
Please don't dispatch unnecessary events - this action is meant to be atomic, once you dispatch it, you want to swap app-db state, re-compute subscriptions and re-render only once, not twice (once with new db as a result of this event and immediately after that, as a result of `:account-update-wallet-settings` event).
https://github.com/status-im/ideas/issues/31 - User story number 3

Reply:
Now, instead of doing the additions dispatch, the whole handler could be just simple db handler:
```clj
(handlers/register-handler-db
  :wallet.settings/toggle-visible-token
  (fn [{:keys [network] :accounts/keys [current-account-id] :as db} [_ symbol checked?]]
    ...as before
    (-> db
        (assoc-in path new-settings)
        (accounts/update-wallet-settings new-settings)))))
```

Reply:
Thanks for the help! It is much better now.